src/wp-includes/rest-api.php

```diff
@@ -2948,6 +2949,14 @@ function rest_preload_api_request( $memo, $path ) {
 		return $memo;
 	}
 
+	if ( isset( $path_parts['path'] ) && '/' !== $path_parts['path'] ) {
+		// Remove trailing slashes from the "path" part of the REST API path.
+		$path_parts['path'] = untrailingslashit( $path_parts['path'] );
+		$path               = str_contains( $path, '?' ) ?
+			$path_parts['path'] . '?' . ( $path_parts['query'] ?? '' ) :
```

Comment:
Thanks for refactoring it, @swissspidy.
LGTM.
Honestly, I didn’t know it’s allowed to use null-coalescing operators, as I was told there is no agreement on using them in Core yet (despite the minimum supported version being PHP 7.2).

Reply:
Right now it‘s a bit of a gray area as a bunch of newer code in core actually uses it.